captum/attr/_core/feature_ablation.py

```diff
@@ -209,16 +225,45 @@ def attribute(
         initial_eval = _run_forward(
             self.forward_func, inputs, target, additional_forward_args
         )
-        assert (
-            initial_eval[0].numel() == 1
-        ), "Target should identify a single element in the model output."
-        initial_eval = initial_eval.reshape(1, num_examples)
+        if (
+            isinstance(initial_eval, int)
+            or isinstance(initial_eval, float)
+            or (
+                isinstance(initial_eval, torch.Tensor)
+                and (
+                    len(initial_eval.shape) == 0
+                    or (num_examples > 1 and initial_eval.numel() == 1)
+                )
+            )
+        ):
+            single_output_mode = True
+            assert (
+                ablations_per_eval == 1
+            ), "Cannot have ablations_per_eval > 1 when function returns scalar."
+            if feature_mask is not None:
+                for single_mask in feature_mask:
+                    assert (
+                        single_mask.shape[0] == 1
+                    ), "Cannot provide multiple masks when function returns a scalar."
+        else:
+            single_output_mode = False
+            assert (
+                isinstance(initial_eval, torch.Tensor) and initial_eval[0].numel() == 1
+            ), "Target should identify a single element in the model output."
+            initial_eval = initial_eval.reshape(1, num_examples)
 
         # Initialize attribution totals and counts
-        total_attrib = [torch.zeros_like(input) for input in inputs]
+        if single_output_mode:
+            total_attrib = [torch.zeros_like(input[0:1]) for input in inputs]
+        else:
+            total_attrib = [torch.zeros_like(input) for input in inputs]
+
         # Weights are used in cases where ablations may be overlapping.
         if self.use_weights:
-            weights = [torch.zeros_like(input) for input in inputs]
+            if single_output_mode:
+                weights = [torch.zeros_like(input[0:1]) for input in inputs]
+            else:
+                weights = [torch.zeros_like(input) for input in inputs]
```

Comment:
nit: perhaps this will be more concise : 
```
weights = [torch.zeros_like(input[0:1] if single_output_mode else input) for input in inputs]
```
same for line 256